From 4eb320802125cc28277ce475d3346a99a3945fe6 Mon Sep 17 00:00:00 2001
From: Alexis Ashley <github@ashley-family.net>
Date: Mon, 8 Sep 2025 10:41:18 +0100
Subject: [PATCH] fix: don't try to compare schemeURI if it is NULL

---
 src/isomedia/isom_write.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/isomedia/isom_write.c b/src/isomedia/isom_write.c
index e8047febd..4993d3b23 100644
--- a/src/isomedia/isom_write.c
+++ b/src/isomedia/isom_write.c
@@ -3129,7 +3129,7 @@ GF_Err gf_isom_add_track_kind(GF_ISOFile *movie, u32 trackNumber, const char *sc
 	}
 
 	map = udta_getEntry(udta, GF_ISOM_BOX_TYPE_KIND, NULL);
-	if (map) {
+	if (map && schemeURI) {
 		count = gf_list_count(map->boxes);
 		for (i=0; i<count; i++) {
 			GF_Box *b = (GF_Box *)gf_list_get(map->boxes, i);
-- 
2.43.0

